### PR TITLE
1485: Improve activation route

### PR DESCRIPTION
**Pull request by f1sh1918**
### Short description

Since we have some issues with associated domains on some android device, it may be useful to also provide a custom scheme link to the user for card activation since it works without domain association

### Proposed changes

- add button for card activation with custom scheme deeplink
- fixed some linting issues

### Side effects

- none 

### Resolved issues

Fixes: #1485 

### Testing is quite difficult on local dev
1. Create a card
2. copy the pdf on your device
3. click on the qr code on the pdf and open with a browser
4. replace the domain with your local network ip to be able to load the local state f.e.
```js
http://192.168.161.22:3000//activation/code#CkkKHwoPQW5kcmVhcyBGaXNjaGVyEO2jARoICgIICSICCAASEAkhZfDlmmOKpEMx5vIR3aoaFKFbplTzRz55MY6S4wA16chnLn5J/
```
5. Switch to "Ehrenamtskarte" (will only appear in dev environment)
6. Click on "Karte aktivieren"

### CODE REVIEW

**Review by ztefanie — APPROVED** [review]
Tested on Android emulator.

I see the screen with the explanation, when I click "Aktivieren", the app opens, but it says the "code is invalid", when i scan the qr code with the camera it says "no connection possible". Although I freshly created the card. This may be not related to this PR, so I approve.

File: administration/src/util/getDeepLinkFromQrCode.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ const getDeepLinkFromQrCode = (qrCode: PdfQrCode): string => {
   const buildConfigProjectId = getBuildConfig(window.location.hostname).common.projectId
   // custom link schemes don't work in browsers or pdf thats why we use the staging link scheme also for development
   const host = isStagingMode() || isDevMode() ? buildConfigProjectId.staging : buildConfigProjectId.production
-  const deepLink = `${HTTPS_SCHEME}://${host}/${ACTIVATION_PATH}/${ACTIVATION_FRAGMENT}${encodeURIComponent(
+  const deepLink = `${HTTPS_SCHEME}://${host}/${ACTIVATION_PATH}/${ACTIVATION_FRAGMENT}#${encodeURIComponent(
```
[the remark this answers is not in the document]
Comment (michael-markl) [reply administration/src/util/getDeepLinkFromQrCode.ts:16]: It should be client side only. There's no reason to send this (private) data to the server for deeplinking.

[the remark this answers is not in the document]
Comment (ztefanie) [reply administration/src/util/getDeepLinkFromQrCode.ts:16]: The activation code is sent to the backend to check if it is valid, isn't it?

**Comment by f1sh1918** [comment]
> Tested on Android emulator.
> 
> I see the screen with the explanation, when I click "Aktivieren", the app opens, but it says the "code is invalid", when i scan the qr code with the camera it says "no connection possible". Although I freshly created the card. This may be not related to this PR, so I approve.

I just tested in again on android real device. Did you replace the link "staging...." with your network ip? And started the emulator with local dev environment? For me everything works fine

https://github.com/digitalfabrik/entitlementcard/assets/37902063/5fe9804b-2c57-4105-9c2a-ffd3bb258336

**Pull request merged**